lips/lip-10.md

```diff
@@ -73,6 +75,32 @@ struct LibraIdDomain {
 * The Libra ID domain can be created as part of creating a `ParentVASP` account.
 * An entity (potentially the Association) may at some point choose to expose an open source application(*Indexer*) that will allow indexing available Libra IDs and could provide a convenient REST API for applications to fetch information about the domains. The API of such an indexer is out of scope of this RFC.
 
+## Libra ID domain events
+
+Blockchain contract that manages Libra ID domains must emit an event every time Libra ID domain is created, removed or updated. Those events are critical for applications to be able to efficiently index existing Libra ID domains.
+
+While Libra ID domains published into VASP accounts, Libra ID domain events are published under the Treasury Compliance account. We consolidate events under single account to allow indexers to follow single event stream.
+
+To store events, `LibraIdDomainManager` resource is published under the Treasure Compliance account(address `0xB1E55ED`).
+
+```
+resource struct LibraIdDomainManager {
+    /// Events are emmitted 
+    libra_id_domain_events: Event::EventHandle<Self::LibraIdDomainEvent>,
+}
+
+struct LibraIdDomainEvent {
+    removed: bool,
+    domain: LibraIdDomain,
+    address: address,
```

Comment:
If it is a domain id update event, we need to know its old domain id, because a parent Vasp can have multiple of domain ids

Reply:
There is no really 'id' of a domain.
The name of domain can be used to identify it, it is part of `LibraIdDomain`.

Reply:
I think my question is:
let's say we update a domain for an address, how do we know the old value that is updated? 
For example, when Novi updates one of its domains "www.novi.com" to "www.novi2.com", emitting the following event does not let people know the old domain (e.g. whether it is updated from "www.novi.com" or "www.novi_sth_else.com" ) 
```
struct LibraIdDomainEvent {
    removed: false,
    domain: "www.novi2.com,
    address: "some_address",
 }
```

Reply:
Oh, I see

I don't think we going to support simple operation of 'updating address'.

In practice, updating address is going to be multi-stage process, that basically consists of introducing new domain, and slowly deprecating old domain, and then finally removing it from the chain

Reply:
In the domain, Lu shows it as www.novi2.com, but wouldn't it be more like just "novi"?

And if there is no updating, then why do we even have update as a type in the "op"?

Reply:
>In the domain, Lu shows it as www.novi2.com, but wouldn't it be more like just "novi"?

Yes, currently domain is something like 'novi', domain regex does not allow `.` in it.

>And if there is no updating, then why do we even have update as a type in the "op"?

There is no updating of domain name(e.g. identifier), but we most likely going to have more fields in `LibraIdDomain`, which can be updated